smsdk/ma_session.py

```diff
@@ -102,6 +102,9 @@ def _get_records(
                     return records
                 _offset += this_loop_limit
 
+            except ValueError as e:
+                raise e
```

Comment:
Is it okay to break complete loop incase any response other than 200, 201. Before this change there is possibility of getting response in next try but with this change It will break complete process. we need to validate it once.

Reply:
Updated the changes and catching not found exceptions instead of just other than [200,201].

Reply:
what if 500 error otherwise looks good to me(LGTM)  ?